fineract-provider/build.gradle

```diff
@@ -278,6 +280,88 @@ configurations {
 tasks.withType(JavaCompile) {
     options.compilerArgs += ["-Xlint:unchecked","-Xlint:cast","-Werror"] // TODO FINERACT-959 (gradually) enable -Xlint:all (see "javac -help -X")
     options.deprecation = true
+    options.errorprone {
+        disableWarningsInGeneratedCode = true
+        //Todo Remove necessary checks from this list
+        disable("UnusedVariable",
+                "VariableNameSameAsType",
+                "InconsistentCapitalization",
+                "UnnecessaryParentheses",
+                "ClassCanBeStatic",
+                "MissingCasesInEnumSwitch",
+                "CompareToZero",
+                "OperatorPrecedence",
+                "MixedMutabilityReturnType",
+                "SameNameButDifferent",
+                "EqualsGetClass",
+                "ReferenceEquality",
+                "StringSplitter",
+                "MissingOverride",
+                "BigDecimalEquals",
+                "ImmutableEnumChecker",
+                "EqualsUnsafeCast",
+                "AssertEqualsArgumentOrderChecker",
+                "BoxedPrimitiveEquality",
+                "BadImport",
+                "UnusedMethod",
+                "ModifiedButNotUsed",
+                "ArgumentSelectionDefectChecker",
+                "JodaPlusMinusLong",
+                "TypeParameterUnusedInFormals",
+                "DoubleBraceInitialization",
+                "UnusedNestedClass",
+                "UndefinedEquals",
+                "JdkObsolete",
+                "OperatorPrecedence",
+                "NarrowingCompoundAssignment"
+        )
+        //TODO gradually enable these checks
+        error(  "DefaultCharset",
```

Comment:
if I understand you correctly, you want to chop this up into multiple PRs by enabling/uncommenting one or a few rules at a time and fixing them?

Reply:
Yeah